application-admintools-default/src/main/java/com/xwiki/admintools/internal/files/resources/LogsDataResource.java

```diff
@@ -168,6 +162,59 @@ public void addZipEntry(ZipOutputStream zipOutputStream, Map<String, String[]> p
         }
     }
 
+    private byte[] getWindowsByteData(int maxLines, ServerInfo usedServer) throws IOException
+    {
+        String directoryPath = usedServer.getLogsFolderPath();
+
+        // Get list of files in the directory
+        File folder = new File(directoryPath);
+        File[] files = folder.listFiles();
+
+        if (files == null) {
+            files = new File[0];
+        }
+        // Filter files starting with the server filter
+        files = Arrays.stream(files).filter(file -> file.getName().startsWith(usedServer.getLogsHint()))
+            .toArray(File[]::new);
+
+        // Sort files in descending order
+        Arrays.sort(files, Comparator.comparing(File::getName).reversed());
+        // Print the sorted files
+        List<String> combinedLogs = new ArrayList<>();
+        for (File file : files) {
+            combinedLogs.addAll(readFileData(maxLines, file));
+            if (writtenLines >= maxLines) {
```

Comment:
You could also initialize it with maxLines and just remove lines from it, to not need `maxLines` after